homeassistant/components/shopping_list/__init__.py

```diff
@@ -53,10 +55,24 @@
 )
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
+async def async_setup(hass, config):
     """Initialize the shopping list."""
 
+    # Only import if we haven't before.
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
+            )
+        )
+
+    return True
+
+
+@asyncio.coroutine
+def async_setup_entry(hass, config_entry):
```

Comment:
config_entry not used --> if second parameter needed : `config_entry = None`, else remove it.

Reply:
Second parameter is indeed needed:
```
Traceback (most recent call last):
  File "/home/tom/Git/home-assistant/homeassistant/config_entries.py", line 214, in async_setup
    result = await component.async_setup_entry(  # type: ignore
TypeError: async_setup_entry() takes 1 positional argument but 2 were given
```